### ISSUE TITLE
Execution History Clean feature doesn't work as expected since Rundeck 3.3.4

### ISSUE THREAD

**Issue by MegaDrive68k** [rundeck/rundeck#6674]
**Describe the bug**

Execution History Clean feature doesn't work since Rundeck 3.3.4, testing deleting execution every 20 seconds, the executions keep on Activity page (Work as expected on Rundeck 3.2.9 until 3.3.3).

**My Rundeck detail**

* Rundeck version: 3.3.4, 3.3.5, 3.3.6, 3.3.7
* install type: WAR
* OS Name/version: Ubuntu 20.10
* DB Type/version: H2

**To Reproduce**

1. Configure a project with the following Execution History Clean config:

![Screenshot_9](https://user-images.githubusercontent.com/51376003/102141216-3b149b00-3e3f-11eb-9d24-10dc2c1f3692.png)

2. Use a job that executes every 5 seconds (to generate executions):

```xml
<joblist>
  <job>
    <defaultTab>nodes</defaultTab>
    <description></description>
    <executionEnabled>true</executionEnabled>
    <id>d97d8e6d-361f-4b0c-b966-5506ed621fa0</id>
    <loglevel>INFO</loglevel>
    <name>HelloWorld</name>
    <nodeFilterEditable>false</nodeFilterEditable>
    <plugins />
    <schedule>
      <dayofmonth />
      <month day='*' month='*' />
      <time hour='*' minute='*' seconds='0/5' />
      <year year='*' />
    </schedule>
    <scheduleEnabled>true</scheduleEnabled>
    <sequence keepgoing='false' strategy='node-first'>
      <command>
        <exec>whoami</exec>
      </command>
    </sequence>
    <uuid>d97d8e6d-361f-4b0c-b966-5506ed621fa0</uuid>
  </job>
</joblist>
```
3. Test on Rundeck 3.3.3 (also tested on 3.2.9), works as expected:

![Screenshot_10](https://user-images.githubusercontent.com/51376003/102141279-52538880-3e3f-11eb-8ed6-48cd13885f87.png)

4. Now, do the same on Rundeck 3.3.4/3.3.5/3.3.6/3.3.7):

![Screenshot_11](https://user-images.githubusercontent.com/51376003/102141571-ca21b300-3e3f-11eb-8b3b-336b25a52985.png)

![Screenshot_8](https://user-images.githubusercontent.com/51376003/102141399-86c74480-3e3f-11eb-9582-d7547bd5b0f9.png)

**Expected behavior**

Execution History Clean feature should work following the cron rules defined.

**Comment by bla-ckbox** [comment]
Rundeck version:  3.3.6
install type: Docker
DB Type/version: mysql:5.7

**Comment by ThorstenPlock** [comment]
Same thing with newest 3.3.8

**Comment by MegaDrive68k** [comment]
Confirmed on 3.3.8 Community version, on Enterprise works well.

**Comment by Domwa** [comment]
> Confirmed on 3.3.8 too, happening only on the Community version.

Did you checked also with Enterprise version?

**Comment by ChBerrich** [comment]
Same thing with newest 3.3.9, any workaround possible?

**Comment by tetsumaki** [comment]
Same here.
3.5.5 -> Bug
Ok... Upgrade.
3.3.9 -> Bug <- Ho no ! :D

**Comment by robmolenaar** [comment]
Same here on 3.3.9. Hope for a solution soon, the database is growing very fast this way.

**Comment by Domwa** [comment]
@robmolenaar you can set up cron entry that uses rundeck-cli:
`rd executions deletebulk -p project_name -m 50 --older "3d" -y 2>&1 >/dev/null`

**Comment by fdevans** [comment]
Will be fixed in 3.4.0

**Comment by syrm** [comment]
Still not fixed in 3.4.x
